web/src/app/admin/admin-api-keys/AdminAPIKeysDrawer.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { useState } from 'react'
+import {
+  ClickAwayListener,
+  Divider,
+  Drawer,
+  Grid,
+  List,
+  ListItem,
+  ListItemText,
+  Toolbar,
+  Typography,
+  Button,
+  ButtonGroup,
+} from '@mui/material'
+import makeStyles from '@mui/styles/makeStyles'
+import { GQLAPIKey } from '../../../schema'
+import AdminAPIKeysDeleteDialog from './AdminAPIKeysDeleteDialog'
+import { Time } from '../../util/Time'
+
+// property for this object
+interface Props {
+  onClose: () => void
+  apiKey: GQLAPIKey
+  setCreate: (param: boolean) => void
+  setAPIKey: (param: GQLAPIKey) => void
+  setOpenActionAPIKeyDialog: (param: boolean) => void
```

Comment:
Can we just store edit and delete as true/false in state and disconnect/remove these?

The dialogs are already modal, so I don't think we need to control that from here

Reply:
The implementation has been updated when Edit and Create dialog has been separated.